js/reference.js

```diff
@@ -322,6 +347,47 @@ var ERMrest = (function(module) {
         },
 
         /**
+         * This is a private funtion that checks the user permissions for modifying the affiliated entity, record or table
+         * Sets a property on the reference object used by canCreate/canUpdate/canDelete
+         * @memberof ERMrest
+         * @private
+         */
+         _checkPermissions: function (permission) {
+            var editCatalog = false,
+                ignoreRecord = false,
+                ignoreUri = module._annotations.IGNORE,
+                metadata = this._meta[permission],
+                users = [];
+
+            var ignoreOnTable = this._table.annotations[ignoreUri];
+            var ignoreOnSchema = this._table.schema.annotations[ignoreUri];
+
+            var ignoreTable = (ignoreOnTable !== undefined && (ignoreOnTable === null || ignoreOnTable === true || ignoreOnTable.indexOf('edit') > -1 || ignoreOnTable.indexOf('entry') > -1));
+            var ignoreSchema = (ignoreOnSchema !== undefined && (ignoreOnSchema === null || ignoreOnSchema === true || ignoreOnSchema.indexOf('edit') > -1 || ignoreOnSchema.indexOf('entry') > -1));
+            if (ignoreTable || ignoreSchema) ignoreRecord = true;
```

Comment:
Should I be checking for these annotations still in ermrestJS but in the canUpdate/canCreate functions?